dolphinscheduler-task-plugin/dolphinscheduler-task-java/src/main/java/org/apache/dolphinscheduler/plugin/task/java/JavaTask.java

```diff
@@ -0,0 +1,438 @@
+            setExitStatusCode(TaskConstants.EXIT_CODE_FAILURE);
+            Thread.currentThread().interrupt();
+        } catch (RunTypeNotFoundException e) {
+            logger.error(e.getMessage());
+            setExitStatusCode(TaskConstants.EXIT_CODE_FAILURE);
+            throw e;
+        } catch (Exception e) {
+            logger.error("java task failed ", e);
+            setExitStatusCode(TaskConstants.EXIT_CODE_FAILURE);
+            throw new TaskException("run java task error", e);
+        }
+    }
+
+    /**
+     * @description: Construct a shell command for the java Run mode
+     * @date: 7/22/22 2:36 AM
+     * @param: []
+     * @return: java.lang.String
+     **/
+    protected String buildJavaCommand() throws Exception {
+        String sourceCode = buildJavaSourceContent();
+        String className = compilerRawScript(sourceCode);
+        StringBuilder builder = new StringBuilder();
+        builder.append(getJavaCommandPath())
+                .append("java").append(" ")
+                .append(buildResourcePath())
+                .append(" ")
+                .append(className).append(" ")
+                .append(javaParameters.getMainArgs().trim()).append(" ")
+                .append(javaParameters.getJvmArgs().trim());
+        return builder.toString();
+    }
+
+    private void setMainJarName() {
+        ResourceInfo mainJar = javaParameters.getMainJar();
+        String resourceName = getResourceNameOfMainJar(mainJar);
+        mainJar.setRes(resourceName);
+        javaParameters.setMainJar(mainJar);
+    }
+
+    /**
+     * @description: Construct a shell command for the java -jar Run mode
+     * @date: 7/22/22 2:36 AM
+     * @param: []
+     * @return: java.lang.String
+     **/
+    protected String buildJarCommand() {
+        String fullName = javaParameters.getMainJar().getResourceName();
+        String mainJarName = fullName.substring(0, fullName.lastIndexOf('.'));
+        mainJarName = mainJarName.substring(mainJarName.lastIndexOf('.') + 1) + ".jar";
+        StringBuilder builder = new StringBuilder();
+        builder.append(getJavaCommandPath())
+                .append("java").append(" ")
+                .append(buildResourcePath()).append(" ")
+                .append("-jar").append(" ")
+                .append(taskRequest.getExecutePath())
+                .append(mainJarName).append(" ")
+                .append(javaParameters.getMainArgs().trim()).append(" ")
+                .append(javaParameters.getJvmArgs().trim());
+        return builder.toString();
+    }
+
+    private String getResourceNameOfMainJar(ResourceInfo mainJar) {
+        if (null == mainJar) {
+            throw new RuntimeException("The jar for the task is required.");
+        }
+
+        return mainJar.getId() == 0
+                ? mainJar.getRes()
+                // when update resource maybe has error
+                : mainJar.getResourceName().replaceFirst("/", "");
```

Comment:
we have slash in class constant, so as
* `PERIOD` for `"."`
* `SPACE` for `" "`
```suggestion
                : mainJar.getResourceName().replaceFirst(Constants.SINGLE_SLASH, "");
```